### PR TITLE
M2m year

### PR DESCRIPTION
**Pull request by pslin1**
Awards and Students now use model YearOfStudy. Awards uses many-to-many, Students uses one-to-one. Deleting and readding a year of study will still maintain order of years in selection. YearOfStudy is also on delete cascade

### CODE REVIEW

File: FSJ_django20_project/FSJ/models_yearofstudy.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	FOURTH = 4
+	FIFTH = 5
+	YEAR_CHOICES = (
+		(FIRST, _('First')),
+		(SECOND, _('Second')),
+		(THIRD, _('Third')),
+		(FOURTH, _('Fourth')),
+		(FIFTH, _('Fifth'))
+	)
+	year = models.IntegerField(choices = YEAR_CHOICES, verbose_name = _("Year of Study"))
```
Comment (dkotowic) [inline FSJ_django20_project/FSJ/models_yearofstudy.py]: Unable to create new years of study outside of these specific 5. No instances of a year of study exists on fresh install, so new objects need to be created but only allowed to be one of these 5 options. Can create multiple instance of year of study objects with the same year. Does not really allow flexibility of creating additionally years of study beyond hard coded values.

File: FSJ_django20_project/FSJ/models_student.py [hunk 2]
```diff
@@ -1 +1 @@
...
    year = models.OneToOneField(YearOfStudy, on_delete=models.CASCADE)

    def user_class(self):
        return "Student"
            
    def get_program(self):
        return self.program
    
    def student_id(self):
        return self.userid
```
Comment (dkotowic) [inline FSJ_django20_project/FSJ/models_student.py]: one to one means only one student may be in, say, second year

Comment (dkotowic) [reply FSJ_django20_project/FSJ/models_student.py]: Do we really want to cascade on delete? I can't see us deleting a year from existence by deleting a student, but likewise can't see us deleting a student when a year gets deleted that they're associated with (perhaps prevent deletion of the year, or perhaps set all students with that year to some default value or allow the value to be null...)

File: FSJ_django20_project/FSJ/models_award.py [hunk 3]
```diff
@@ -1,35 +1,21 @@
...
-		(FOURTH, _('Fourth')),
-		(FIFTH, _('Fifth'))
-	)
 	awardid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
 	award_name = models.TextField(verbose_name = _("Award Name"))
 	description = models.TextField(verbose_name = _("Description"))
 	value = models.TextField(verbose_name = _("Value"))
 	programs = models.TextField(verbose_name = _("Programs"))
-	years_of_study = models.IntegerField(choices = YEAR_CHOICES, verbose_name = _("Years of Study"))
+	years_of_study = models.ManyToManyField(YearOfStudy)
```
Comment (dkotowic) [inline FSJ_django20_project/FSJ/models_award.py]: May wish to define a behaviour for on delete here too? If an award is deleted, what happens to the years (probably nothing) but if a year that's associated with an award if deleted?

**Comment by pslin1** [comment]
Changes have been adressed except for creating initial values of years of study. It looks like we'll have to modify the migration in order to put in initial values, this might be something to towards the end when we know for sure what our db will look like so we don't have to worry about changing the migration everytime.

**Pull request merged**